client/landing/stepper/hooks/use-site.ts

```diff
@@ -7,21 +7,16 @@ import type { SiteSelect } from '@automattic/data-stores';
 export function useSite() {
 	const siteSlug = useSiteSlugParam();
 	const siteIdParam = useSiteIdParam();
-	const siteId = useSelect(
-		( select ) => siteSlug && ( select( SITE_STORE ) as SiteSelect ).getSiteIdBySlug( siteSlug ),
-		[ siteSlug ]
-	);
+
 	const site = useSelect(
-		( select ) =>
-			( siteId || siteIdParam ) &&
-			( select( SITE_STORE ) as SiteSelect ).getSite(
-				( siteId ?? siteIdParam ) as string | number
-			),
-		[ siteId, siteIdParam ]
+		( select ) => {
+			const siteStore = select( SITE_STORE ) as SiteSelect;
+			const siteKey = siteSlug ?? siteIdParam;
+
+			return siteKey ? siteStore.getSite( siteKey as string | number ) : null;
+		},
+		[ siteSlug, siteIdParam ]
 	);
 
-	if ( ( siteSlug || siteIdParam ) && site ) {
-		return site;
-	}
-	return null;
```

Comment:
We might still need that in order to either return a site object or `null`; otherwise we may end up returning `undefined`, which may be unexpected in some areas.

Reply:
Thanks @tyxla, I rollback these lines [here](https://github.com/Automattic/wp-calypso/pull/77030/commits/6862cd97d3b1b5d1c5a546a1f149c812eb848158)